packages/roosterjs-editor-dom/lib/utils/setColor.ts

```diff
@@ -34,5 +51,105 @@ export default function setColor(
                 element.dataset[dataSetName] = modeIndependentColor.lightModeColor;
             }
         }
+
+        if (isBackgroundColor && shouldAdaptTheFontColor) {
+            adaptFontColorToBackgroundColor(element, isDarkMode);
+        }
     }
 }
+
+/**
+ * Change the font color to white or some other color, so the text can be visible with a darker background
+ * @param element The element that contains text.
+ */
+function adaptFontColorToBackgroundColor(element: HTMLElement, isDarkMode?: boolean) {
+    if (element.firstElementChild?.hasAttribute('style')) {
+        return;
+    }
+    const backgroundColor = element.style.getPropertyValue('background-color');
+    const lightModeBackgroundColor =
+        (isDarkMode &&
+            (element.dataset[DarkModeDatasetNames.OriginalStyleBackgroundColor] ||
+                element.dataset[DarkModeDatasetNames.OriginalAttributeBackgroundColor])) ||
+        backgroundColor;
+    if (!lightModeBackgroundColor || lightModeBackgroundColor === TRANSPARENT) {
+        return;
+    }
+    const isADarkOrBrightOrNone = isADarkOrBrightColor(lightModeBackgroundColor!);
+    switch (isADarkOrBrightOrNone) {
+        case ColorTones.DARK:
+            const fontForDark: ModeIndependentColor = {
+                lightModeColor: WHITE,
+                darkModeColor: GRAY,
+            };
+            setColor(element, fontForDark, false /*isBackground*/, isDarkMode);
+            break;
+        case ColorTones.BRIGHT:
+            const fontForLight: ModeIndependentColor = {
+                lightModeColor: GRAY,
```

Comment:
When background color is bright, should text color be BLACK or GRAY in light mode? I think it should be BLACK (the default value).

Other colors look fine.